lib/extensions/ar_table_lock.rb

```diff
@@ -0,0 +1,35 @@
+module ArTableLock
+  # Creates mutex by locking on a database table in SQL.
+  # Please use record level locking over this
+  #
+  # passing in mode has been disabled due to brakeman
+  #
+  # if a different mode is desired, we'll work around issue later
+  #
+  # possible modes:
+  # "ACCESS SHARE", "ROW SHARE", "ROW EXCLUSIVE", "SHARE", "EXCLUSIVE",
+  #  "SHARE UPDATE EXCLUSIVE", "SHARE ROW EXCLUSIVE", "ACCESS EXCLUSIVE"
+  #
+  # details on locks can be found on postgres docs:
+  #   can be fount http://www.postgresql.org/docs/9.4/static/explicit-locking.html
+  #
+  def with_lock(mode = :shared_update_exclusive, timeout = 60.seconds)
+    lock = case mode
+    when :shared_update_exclusive, :SUE then "SHARE UPDATE EXCLUSIVE"
+    else raise "unknown lock mode <#{mode.inspect}>"
+    end
```

Comment:
This lock variable never gets used and then you overwrite the mode anyway two lines later...is this just for sanity checking the lock mode?